janusgraph-core/src/main/java/org/janusgraph/util/system/ExecuteUtil.java

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.util.system;
+
+import org.apache.commons.lang.UnhandledException;
+import org.janusgraph.diskstorage.BackendException;
+import org.janusgraph.util.datastructures.ExceptionWrapper;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class ExecuteUtil {
+
+    public static void executeWithCatching(ExecuteExceptionallyFunction function, ExceptionWrapper exceptionWrapper){
+        try {
+            function.execute();
+        } catch (Throwable throwable){
+            if(exceptionWrapper.getThrowable() == null){
+                exceptionWrapper.setThrowable(throwable);
+            } else {
+                exceptionWrapper.getThrowable().addSuppressed(throwable);
+            }
+        }
+    }
+
+    public static void throwIfException(ExceptionWrapper exceptionWrapper) throws BackendException {
+        if(exceptionWrapper.getThrowable() != null){
+            Throwable throwable = exceptionWrapper.getThrowable();
+            if(throwable instanceof BackendException){
+                throw (BackendException) throwable;
+            } else if(throwable instanceof RuntimeException){
+                throw (RuntimeException) throwable;
+            } else {
+                throw new UnhandledException(throwable);
+            }
+        }
+    }
+
+    public static void gracefulExecutorServiceShutdown(ExecutorService executorService, long maxWaitTime){
+        if(executorService == null){
```

Comment:
We can do
```suggestion
        if(executorService == null || !executorService.isShutdown()){
```
to make this function idempotent.

Reply:
I think it should be `if(executorService == null || !executorService.isTerminated()())` because if it isn't terminated in specified amount of time, the underlying threads of executor service should be interrupted. Changed it